### PR TITLE
Update codeowner and common file path

### PR DESCRIPTION
**Pull request by RissyRan**
# Description

Update codeowner and common file path:
* Code owner file: update the new owner and remove folks who are not in the team
* Move commonly used `test_owner`, `vm_resources` and `quarantined_tests` files into a `common` directory and add owner accordingly, so changes in that file will not be blocked by the review of infra owners

# Tests

Please describe the tests that you ran on Cloud VM to verify changes.

**Instruction and/or command lines to reproduce your tests:** ...
Upload to airflow and no compiler error & github runner passes

**List links for your tests (use go/shortn-gen for any internal link):** ...
No import errors - [SS link](https://screenshot.googleplex.com/4jdW8A86sSCR9Ti)

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

**Comment by RissyRan** [comment]
Hi @yixinshi @allenwang28 @zpcore @gobbleturk @shralex @jiangjy1982 @ortibazar @parambole @vipannalla @crankshaw-google, could you please take a look and see if you are the right owner for `dags/common` directory? It includes files `test_owner`, `vm_resources` and `quarantined_tests`. Thank you!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,20 +1,22 @@
 # Default owners for everything in the repo, unless a later match takes precedence.
-* @mbzomowski @RissyRan @allenwang28
+* @yixinshi @RissyRan @allenwang28
```
Comment (richardsliu) [inline .github/CODEOWNERS:2]: Can you add me here?

**Review by shralex — APPROVED** [review]

**Review by richardsliu — APPROVED** [review]

**Review by vipannalla — APPROVED** [review]
LGTM, thanks for making these changes.

**Review by JackCaoG — APPROVED** [review]

**Review by ManfeiBai — APPROVED** [review]
Thanks, LGTM

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,20 +1,22 @@
 # Default owners for everything in the repo, unless a later match takes precedence.
-* @mbzomowski @RissyRan @allenwang28
+* @richardsliu @yixinshi @RissyRan @allenwang28
+
+dags/common @richardsliu @yixinshi @RissyRan @allenwang28 @zpcore @ManfeiBai @gobbleturk @shralex @jiangjy1982 @ortibazar @parambole @vipannalla @crankshaw-google
```
Comment (crankshaw-google) [inline .github/CODEOWNERS:4]: please add @polydier1 from my team as well

**Review by crankshaw-google — APPROVED** [review]
LGTM

File: .github/CODEOWNERS (hunk 2 above)
Comment (RissyRan) [reply .github/CODEOWNERS]: Np, added!

**Review by ortibazar — APPROVED** [review]

**Review by gobbleturk — APPROVED** [review]

**Review by parambole — APPROVED** [review]
LGTM.

**Pull request merged**